### DISCUSSION TITLE
lensfun 0.3.95 alpha release breaks backwards compatibility

### DISCUSSION THREAD

**Discussion by apfelkraut** [Homebrew/discussions#3070] [Tap maintenance and Homebrew development]
Homebrew currently provides an "alpha" (in contrast to stable) version of lensfun: _"2018-06-29 Alpha release 0.3.95: This is an alpha release and snapshot of the current development."_, see https://lensfun.github.io (its original homepage) for details.

The major issue about lensfun 0.3.95 is that it breaks backward compatibility (see https://github.com/lensfun/lensfun/issues/945) to previous stable releases. The effect is that in case you edited your photos with a previous version, those edits will be broken and you can simply start from scratch with all of your work.

Consumers of lensfun like [darktable.org](https://www.darktable.org/) have also reported about this issue, see https://github.com/darktable-org/darktable/issues/2813. Due to that they are still making use of the last stable release of lensfun which is 0.3.2 (dating back to 2015). macports also has an issue about this aspect: https://trac.macports.org/ticket/64024.

Happily lensfun just released a stable successor of 0.3.2 in form of https://github.com/lensfun/lensfun/releases/tag/v0.3.3 (as fresh as 2022) . So this one is build on top of 0.3.2 and incorporates all the new lenses and cameras but not the changes that broke backward compatibility in alpha 0.3.95.

To all what I found out the current lensfun formula is not a dependency to any other package.

I would suggest to change the default `lensfun` formula to provide the latest stable release v0.3.3. In addition the current formula could be renamed to `lensfun@0.3.95`, in case it is (still) of any use to someone.

I would love to help, but I am currently not even able to build the existing formula from source to have a working version of lensfun alpha 0.3.95 (https://github.com/Homebrew/homebrew-core/issues/96662).

Any feedback or help would be highly appreciated! Thanks a lot!

**Comment by SMillerDev** [comment]
This really seems like something that would be easier to discuss in a pull request. Would you mind making one?

**Reply by apfelkraut** [reply]
Thanks @SMillerDev. I would be happy to provide one including these changes. The problem is that although the build from sources works, it results in a version of lensfun that does not work on my machine at least (see https://github.com/Homebrew/homebrew-core/issues/96662) ... shall I still do so?

**Reply by SMillerDev** [reply]
Preferably you'd consult with upstream first to fix the broken build, but if that's not possible making a pull request is fine.

**Reply by apfelkraut** [reply]
My issue to fix the broken build was immediately closed. I have just created a pull request with the changes: https://github.com/Homebrew/homebrew-core/pull/96709

**Reply by SMillerDev** [reply]
If you mean the issue you linked above, that's not with upstream though. That is with homebrew.
